### ISSUE TITLE
Maximum call stack size exceeded -> In SourceMapDevToolPlugin/ Tappable

### ISSUE THREAD

**Issue by tmcconechy** [webpack/webpack#16658]
# Bug report

**What is the current behavior?**

When running the dev-server in live-reload mode and changing files the dev server will immediately die with this stack trace when changing a file. With this stack trace:

```
RangeError: Maximum call stack size exceeded
    at eval (eval at create (/Users/tim.mcconechy/Dev/enterprise-wc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:11:1)
    at /Users/tim.mcconechy/Dev/enterprise-wc/node_modules/webpack/lib/SourceMapDevToolPlugin.js:549:10
    at /Users/tim.mcconechy/Dev/enterprise-wc/node_modules/neo-async/async.js:2830:7
    at Object.each (/Users/tim.mcconechy/Dev/enterprise-wc/node_modules/neo-async/async.js:2857:9)
    at /Users/tim.mcconechy/Dev/enterprise-wc/node_modules/webpack/lib/SourceMapDevToolPlugin.js:384:17
    at /Users/tim.mcconechy/Dev/enterprise-wc/node_modules/neo-async/async.js:2830:7
    at Object.each (/Users/tim.mcconechy/Dev/enterprise-wc/node_modules/neo-async/async.js:2857:9)
    at /Users/tim.mcconechy/Dev/enterprise-wc/node_modules/webpack/lib/SourceMapDevToolPlugin.js:207:15
    at /Users/tim.mcconechy/Dev/enterprise-wc/node_modules/webpack/lib/Compilation.js:501:18
    at Hook.eval [as callAsync] (eval at create (/Users/tim.mcconechy/Dev/enterprise-wc/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:7:1)
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Can see it in our repo
-  clone https://github.com/infor-design/enterprise-wc
-  edit `webpack.config.dev.js` so that  `devtool: 'cheap-module-source-map' ` (not `eval-source-map` as it is now as a workaround)
- `nvm use` (node 18 or 16 so not a node issue)
- `npm i` 
- `npm run start`
- change any file like src/components/ids-about/ids-about.ts (remove a `;`) and press save
- stack trace will appear

**What is the expected behavior?**
This seems to be a new issue with something introduced recently in the `npm` ecosystem, as it happens on old and new versions of our project with various versions of the dependencies.

devtool seems related ->https://webpack.js.org/configuration/devtool/ 
I also changed some of these options in webpack.config.dev.js and the "eval" ones seem to work...

Seems also to happen with more files like our repo has. We can run in "filter mode" with less entries and the issue does not occur `npm --components=about run start`

**Other relevant information:**
webpack version:  5.75.0
Node.js version: 18
Operating System: Mac
Additional tools:

**Comment by alexander-akait** [comment]
Looks like some source maps is really deep to merge... I will look deeply soon

**Comment by TheLarkInn** [comment]
@tmcconechy would you be willing to test this on the latest version of webpack?

**Comment by tmcconechy** [comment]
@TheLarkInn no it does not seem to be fixed on the latest.

**Comment by tmcconechy** [comment]
@alexander-akait Any updates or other reports of this. Seems like now i get this everytime and none of the workarounds work anymore. But the repo is now moved privately so sort of just asking if any similar issues with solutions?

**Comment by alexander-akait** [comment]
@tmcconechy hm, I haven't received any messages about this issue for a long time, and I assumed that it might have been fixed since then, but apparently not, is it possible to somehow try to get it from a private repository? Get a longer stack/more information on which file this happens, in what place?

**Comment by alexander-akait** [comment]
Yeah, looks like you have very deep source maps, I want to say it is a limitation and it requires to increase stack size...

**Comment by tmcconechy** [comment]
@alexander-akait that must be it because this also seemed to happen as the code base got bigger and bigger. Lets go with the stack size as a solution. Now its noted here in case someone needs it

**Comment by rilpires** [comment]
Having this problem with papaparse.min.js
https://github.com/mholt/PapaParse/issues/1118
Not sure what happened since last friday (6th March)
Just increasing the node stack-size does not work. Using `--stack-size=5000` still triggers the error, and reaching `--stack-size=10000` throws a Segmentation fault error

**Comment by alexander-akait** [comment]
@rilpires Can you create a simple reproducible test repo?

**Comment by dyogenkumar** [comment]
I noticed that in my case, after increment Angular 20.3.16 to 20.3.17, getting error. Which was again due to newer version of copy-webpack-plugin@14. earlier it was using copy-webpack-plugin@13.0.1

**Comment by rilpires** [comment]
> [@rilpires](https://github.com/rilpires) Can you create a simple reproducible test repo?

I've tried a vanilla angular project that imports papaparse and could manage to build it, so I'll work on bisecting the actual culprit

**Comment by alexander-akait** [comment]
@rilpires Feel free to ping me

**Comment by asgarov1** [comment]
Happened to me recently when I patched our application to Angular v20.3.18:
```
./node_modules/papaparse/papaparse.min.js - Error: Module build failed (from ./node_modules/@angular-devkit/build-angular/src/tools/babel/webpack-loader.js):                                                              
RangeError: /home/.../node_modules/papaparse/papaparse.min.js: Maximum call stack size exceeded 
```

Solution was to update the version of **@babel/helper-define-polyfill-provider** to **v0.6.8**+

**Comment by hai-x** [comment]
Yeah, it looks like the parsing issue with `papaparse.min.js` has been fixed [on the Babel side](https://github.com/babel/babel-polyfills/issues/254).

**Comment by derekblackwood61-lang** [comment]
Thanks for confirming! Updating @babel/helper-define-polyfill-provider to
v0.6.8+ fixed the stack overflow issue for me as well after upgrading to
Angular v20.3.18.
